### PR TITLE
clean up controller package

### PR DESCRIPTION
**Pull request by shanalily**
There's currently a lot going on in the controller package that isn't specifically related to k8s controllers.

Summary of original:

```
controller/
    nodelabel_controller.go
    computeresource.go
    config.go
    naming.go
azure/
    *misc. stuff*
```

Changed to:

```
controller/
    nodelabel_controller.go
    filter.go
labelsync/
    labelsync.go
    patch.go
    options/
        config.go
    naming/
        naming.go
azure/
    *misc. stuff*
    computeresource/
        computeresource.go
        vm.go
        vmss.go
```

### CODE REVIEW

File: controller/filter.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+func timeToUpdate(node *corev1.Node) bool {
+	label, ok := node.Labels[lastUpdateLabel]
+	if !ok {
+		return true // let things through the first time
+	}
+	var period time.Duration
+	// if lastUpdate formatted incorrectly, do I let stuff through?
+	lastUpdate, err := time.Parse(time.RFC3339, strings.ReplaceAll(label, ".", ":"))
+	if err != nil {
```
Comment (serbrech) [inline controller/filter.go]: Should you log something for the user to figure out his mistake, and correct it?

Comment (serbrech) [reply controller/filter.go]: different PR though. keep this one focused on package structure.

File: controller/nodelabel_controller.go [hunk 2]
```diff
@@ -54,11 +52,11 @@ func (r *ReconcileNodeLabel) Reconcile(req reconcile.Request) (reconcile.Result,
 	log := r.Log.WithValues("node-label-operator", req.NamespacedName)
 
 	var configMap corev1.ConfigMap
-	optionsNamespacedName := OptionsConfigMapNamespacedName() // assuming "node-label-operator" and "node-label-operator-system", is this okay
+	optionsNamespacedName := options.OptionsConfigMapNamespacedName() // assuming "node-label-operator" and "node-label-operator-system", is this okay
```
Comment (serbrech) [inline controller/nodelabel_controller.go]: `options.ConfigMapNamespaceName()` remove the Options prefix as it is now in the options package

File: controller/filter.go (hunk 1 above)
Comment (shanalily) [reply controller/filter.go]: Yeah I probably should log something.

File: controller/nodelabel_controller.go [hunk 3]
```diff
@@ -54,11 +52,11 @@ func (r *ReconcileNodeLabel) Reconcile(req reconcile.Request) (reconcile.Result,
...
 	if err := r.Get(r.ctx, optionsNamespacedName, &configMap); err != nil {
 		log.V(1).Info("unable to fetch ConfigMap, instead using default configuration settings")
 		// create default options config map and requeue
-		configMap, err := NewDefaultConfigOptions()
+		configMap, err := options.NewDefaultConfigOptions()
```
Comment (serbrech) [inline controller/nodelabel_controller.go]: `options.NewDefault()` ?

File: controller/nodelabel_controller.go [hunk 4]
```diff
@@ -69,7 +67,7 @@ func (r *ReconcileNodeLabel) Reconcile(req reconcile.Request) (reconcile.Result,
 		}
 		return ctrl.Result{RequeueAfter: time.Minute}, nil
 	}
-	configOptions, err := NewConfigOptions(configMap)
+	configOptions, err := options.NewConfigOptions(configMap)
```
Comment (serbrech) [inline controller/nodelabel_controller.go]: options.New() options.NewConfig()?

**Pull request merged**